torch/csrc/distributed/c10d/ProcessGroupUCC.cpp

```diff
@@ -128,11 +128,48 @@ std::map<std::string, std::string> torch_ucc_envs_map = {
     {"TORCH_UCC_ENABLE_COMMS_LOGGER", "0"},
 };
 
+std::unordered_set<OpType> parse_blocking_wait(std::string s) {
+  const static std::unordered_map<std::string, OpType> str2op = {
+      {"allgather", OpType::ALLGATHER},
+      {"allgather_base", OpType::_ALLGATHER_BASE},
+      {"allreduce", OpType::ALLREDUCE},
+      {"alltoall_base", OpType::ALLTOALL_BASE},
+      {"broadcast", OpType::BROADCAST},
+      {"gather", OpType::GATHER},
+      {"reduce", OpType::REDUCE},
+      {"reduce_scatter", OpType::REDUCE_SCATTER},
+      {"scatter", OpType::SCATTER},
+      {"send", OpType::SEND},
+      {"recv", OpType::RECV},
+  };
+  std::unordered_set<OpType> result;
+  s = tolower(trim(s));
+  if (s == "0") {
+    return result;
+  }
+  if (s == "1" || s == "all") {
+    result.reserve(str2op.size());
+    for (auto item : str2op) {
+      result.insert(item.second);
+    }
+    return result;
+  }
+  while (!s.empty()) {
+    const auto end_pos = s.find_first_of(',');
+    const auto token = trim(s.substr(0, end_pos));
+    const auto it = str2op.find(std::string(token));
+    TORCH_CHECK(it != str2op.end(), "Invalid blocking wait option: '", token);
```

Comment:
using str2op.at() can be cleaner and automatically handles the exception.

Reply:
done